templates/pages/reset-password.jsx

```diff
@@ -6,18 +6,29 @@ var ResetView = require('../components/reset-password-view.jsx');
 var RequestView = require('../components/request-reset-view.jsx');
 var Router = require('react-router');
 
+
+var Url = require('url');
+require('whatwg-fetch');
+
 // This wraps every view
 var ResetPassword = React.createClass({
   getInitialState: function() {
+    var queryObj = Url.parse(window.location.href, true).query;
```

Comment:
This will be problematic when we do test with no `window` context, but we can deal with this later when we add the test.